rewrite-maven/src/main/java/org/openrewrite/maven/tree/ResolvedPom.java

```diff
@@ -440,35 +456,36 @@ private void resolveParentPropertiesAndRepositoriesRecursively(List<Pom> pomAnce
         }
 
         private void resolveParentDependenciesRecursively(List<Pom> pomAncestry) throws MavenDownloadingException {
-            if(pomAncestry.isEmpty()) {
+            if (pomAncestry.isEmpty()) {
                 return;
             }
-            Map<GroupArtifactScope, ResolvedManagedDependencyDepth> gaToNearest = dependencyManagement.stream()
+            Map<GroupArtifactClassifierType, ResolvedManagedDependencyDepth> managedDependencyMap = dependencyManagement.stream()
```

Comment:
As can be seen here the managed dependencies are now stored in a map, where `(groupId, artifactId, classifier, type)` is used as key rather than `GroupArtifactScope`, which also included the scope (`(groupId, artifactId, classifier, type, scope)`), which was wrong.